src/Material.cpp

```diff
@@ -1259,13 +1342,69 @@ void Material::setDifTildeByGroup(double xs, int group, int surface) {
 
 
 /**
+ * @brief Set the Material's array of diffusion coefficients.
+ * @details This method is a helper function to allow OpenMOC users to assign
+ *          the Material's nuclear data in Python. A user must initialize a
+ *          NumPy array of the correct size (i.e., a float64 array the length
+ *          of the number of energy groups) as input to this function. This
+ *          function then fills the NumPy array with the data values for the
+ *          Material's buckling coefficients. An example of how this function
+ *          might be called in Python is as follows:
+ *
+ * @code
+ *          buckling = numpy.array([0.05, 0.1, 0.15, ... ])
+ *          material = openmoc.Material(openmoc.material_id())
+ *          material.setBuckling(buckling)
+ * @endcode
+ *
+ * @param xs the array of diffusion coefficents
+ * @param num_groups the number of energy groups
+ */
+void MacroMaterial::setBuckling(double* xs, int num_groups) {
+
+  if (_num_groups != num_groups)
+    log_printf(ERROR, "Unable to set diffusion coefficient with %d groups for "
+               "Material %d which contains %d energy groups", num_groups,
+               _id, _num_groups);
+
+  if (_buckling == NULL)
+    _buckling = new FP_PRECISION[_num_groups];
+
+  for (int i=0; i < _num_groups; i++)
+    _buckling[i] = xs[i];
+}
+
+
+/**
+ * @brief Set the Material's diffusion coefficient for some energy group.
+ * @param xs the diffusion coefficient
+ * @param group the energy group
+ */
+void MacroMaterial::setBucklingByGroup(double xs, int group) {
+
+  if (group <= 0 || group > _num_groups)
+    log_printf(ERROR, "Unable to set diffusion coefficient for group %d for "
+               "Material %d which contains %d energy groups",
+               group, _id, _num_groups);
+
+  if (_buckling == NULL){
+    _buckling = new FP_PRECISION[_num_groups];
+    for (int i=0; i < _num_groups; i++)
+      _buckling[i] = 0.0;
+  }
+
+  _buckling[group-1] = xs;
+}
+
+
+/**
  * @brief Checks if the total cross-section for this Material is equal to the
  *        absorption plus scattering cross-sections for all energy groups.
  * @details If the total cross-section does not equal the absorption plus
  *          scattering cross-section within SIGMA_T_THRESH then this method
  *          exits OpenMOC.
  */
-void Material::checkSigmaT() {
+void MacroMaterial::checkSigmaT() {
 
```

Comment:
Do you think we should make the message in ``MacroMaterial::checkSigmaT()`` a ``WARNING`` message instead of an ``ERROR`` message? This seems to be more inline with where you and I are taking the code, as well was with PR #115's inclusion of code to account for nu-scatter.